Doppler.ContactPolicies.Data.Access/Repositories/ContactPoliciesSettings/ContactPoliciesSettingsRepository.cs

```diff
@@ -57,34 +54,34 @@ public async Task UpdateContactPoliciesSettingsAsync(string accountName, Entitie
             const string updateQuery =
                 @"update [UserShippingLimit] set Active = @Active, Interval = @IntervalInDays, Amount = @EmailsAmountByInterval
                 from [UserShippingLimit] usl
-                inner join [User] u on u.IdUser = usl.IdUser where u.Email = @Email and usl.Enabled = 1;";
+                inner join [User] u on u.IdUser = usl.IdUser where u.IdUser = @IdUser and usl.Enabled = 1;";
 
             var affectedRows = await connection.ExecuteAsync(updateQuery, new
             {
-                Email = accountName,
+                IdUser = idUser,
                 contactPoliciesToInsert.IntervalInDays,
                 contactPoliciesToInsert.EmailsAmountByInterval,
                 contactPoliciesToInsert.Active
             }, transaction);
 
             if (affectedRows == 0)
-                throw new Exception($"This action is not allowed for the user with Account {accountName}.");
+                throw new Exception($"This action is not allowed for this user.");
 
             await connection.ExecuteAsync(
-                @"delete [SubscribersListXShippingLimit] from [SubscribersListXShippingLimit] slxsl inner join [User] u on u.IdUser = slxsl.IdUser where u.Email = @Email and  IdSubscribersList not in @Ids;",
-                new { Email = accountName, Ids = contactPoliciesToInsert.ExcludedSubscribersLists.Select(x => x.Id) }, transaction);
+                @"delete [SubscribersListXShippingLimit] from [SubscribersListXShippingLimit] slxsl inner join [User] u on u.IdUser = slxsl.IdUser where u.IdUser = @IdUser and  IdSubscribersList not in @Ids;",
+                new { IdUser = idUser, Ids = contactPoliciesToInsert.ExcludedSubscribersLists.Select(x => x.Id) }, transaction);
```

Comment:
Now you can remove the join with `User` table, right?

```suggestion
                @"delete [SubscribersListXShippingLimit] from [SubscribersListXShippingLimit] slxsl where slxsl.IdUser = @IdUser and IdSubscribersList not in @Ids;",
                new { IdUser = idUser, Ids = contactPoliciesToInsert.ExcludedSubscribersLists.Select(x => x.Id) }, transaction);
```

Reply:
👍